File: app/src/main/java/com/example/tfg_urjc_appfirstrun/Database/Labs/TrainingLab.java
package com.example.tfg_urjc_appfirstrun.Database.Labs;

import android.annotation.SuppressLint;
import android.content.Context;

import androidx.room.Room;

import com.example.tfg_urjc_appfirstrun.DAO.TrainingDao;
import com.example.tfg_urjc_appfirstrun.Database.TrainingDatabase;
import com.example.tfg_urjc_appfirstrun.Entities.Training;

import java.util.List;

public class TrainingLab {
    @SuppressLint("StaticFieldLeak")
    private static TrainingLab sTrainingLab;

    private TrainingDao mTrainingDao;

    private TrainingLab(Context context) {
        Context appContext = context.getApplicationContext();
        TrainingDatabase db = Room.databaseBuilder(appContext,
                TrainingDatabase.class, "tfg_database").build();
        mTrainingDao = db.trainingDao();
    }

    public static TrainingLab get(Context context) {
        if (sTrainingLab == null) {
            sTrainingLab = new TrainingLab(context);
        }
        return sTrainingLab;
    }

    public List<Training> getTraining() {
        return mTrainingDao.getAll();
    }

    public Training getTrainingById(String id) {
        return mTrainingDao.getById(id);
    }

    public void addTraining(Training training) {
        mTrainingDao.insert(training);
    }

    /*public void updateTraining(Training training) {
        mNotaDao.updateNota(training);
    }*/

    public void deleteTraining(Training training) {
        mTrainingDao.delete(training);
    }
}

File: app/src/main/java/com/example/tfg_urjc_appfirstrun/Database/Labs/TrainingLab.kt
package com.example.tfg_urjc_appfirstrun.Database.Labs

import android.annotation.SuppressLint
import android.content.Context
import androidx.room.Room
import com.example.tfg_urjc_appfirstrun.DAO.TrainingDao
import com.example.tfg_urjc_appfirstrun.Database.TrainingDatabase
import com.example.tfg_urjc_appfirstrun.Entities.Training

class TrainingLab private constructor(context: Context?) {
    private val mTrainingDao: TrainingDao?
    fun getTraining(): List<Training?>? {
        return mTrainingDao?.getAll()
    }

    suspend fun getTrainingById(id: String?): Training? {
        return mTrainingDao?.getById(id)
    }

    fun getActualTraining(): Training? {
        return mTrainingDao?.getActualTraining()
    }

    suspend fun addTraining(training: Training?) {
        mTrainingDao?.insert(training)
    }

    /*public void updateTraining(Training training) {
        mNotaDao.updateNota(training);
    }*/
    suspend fun deleteTraining(training: Training?) {
        mTrainingDao?.delete(training)
    }

    companion object {
        @SuppressLint("StaticFieldLeak")
        private var sTrainingLab: TrainingLab? = null
        operator fun get(context: Context?): TrainingLab? {
            if (sTrainingLab == null) {
                sTrainingLab = TrainingLab(context)
            }
            return sTrainingLab
        }
    }

    init {

        val appContext = context!!.applicationContext
        val database: TrainingDatabase = Room.databaseBuilder(
                appContext,
                TrainingDatabase::class.java, "tfg-urjc-strava"
        )
        .allowMainThreadQueries()
        .build()
        mTrainingDao = database.trainingDao()
    }
}